lib/Backends/CPU/CPUDeviceManager.cpp

```diff
@@ -32,7 +32,9 @@ static llvm::cl::opt<unsigned, /* ExternalStorage */ true> GlowCPUMemoryOpt(
 DeviceManager *createCPUDeviceManager(const DeviceConfig &config) {
   if (GlowCPUMemory) {
     // Convert command line GlowCPUMemory to bytes from kilobytes.
-    return new CPUDeviceManager(config, uint64_t{GlowCPUMemory} * 1024);
+    auto configNew = DeviceConfig(BackendKind::CPU);
+    configNew.setDeviceMemory(uint64_t{GlowCPUMemory} * 1024);
```

Comment:
we only need to do so if the 
```
const DeviceConfig &config)
```
does not have a device memory set, right?

Reply:
> we only need to do so if the
> 
> ```
> const DeviceConfig &config)
> ```
> 
> does not have a device memory set, right?
> 
> It also not safe to create new device config this way. Say new param is added to the config, and now we start missing that param. We need to create new config based on the copy ctor and redefine device memory.

If I directly call the config->setDeviceMemory() , it will cause the error bellow:
 error: 'this' argument to member function 'setDeviceMemory' has type 'const glow::runtime::DeviceConfig', but function is not marked const
    config.setDeviceMemory(uint64_t{GlowCPUMemory} * 1024);

Is that because the argument is const and I should not change it?  Should I simply create new config based on the copy ctor?

Reply:
> config->setDeviceMemory()

cannot be used because config is a const reference. This is ok.

> Should I simply create new config based on the copy ctor?

this seems reasonable.

Reply:
> > config->setDeviceMemory()
> 
> cannot be used because config is a const reference. This is ok.
> 
> > Should I simply create new config based on the copy ctor?
> 
> this seems reasonable.

OK. I switched to using copy ctor to create new config. Thanks :)